pkg/otlp/model/translator/testhelper_test.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package translator
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/quantile"
+	"github.com/DataDog/datadog-agent/pkg/quantile/summary"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// TestMetrics is the struct used for serializing Datadog metrics for generating testdata.
+// It contains sketches (distributions) and timeseries (all other types).
+// This structure is not meant to be used directly; use AssertTranslatorMap instead.
+type TestMetrics struct {
+	Sketches   []TestSketch
+	TimeSeries []TestTimeSeries
+}
+
+// TestDimensions copies the Dimensions struct with public fields.
+// NOTE: Keep this in sync with the Dimensions struct.
+type TestDimensions struct {
+	Name     string
+	Tags     []string
+	Host     string
+	OriginID string
+}
+
+type TestSketch struct {
+	TestDimensions
+	Timestamp uint64
+	Summary   summary.Summary
+	Keys      []int32
+	Counts    []uint32
+}
+
+type TestTimeSeries struct {
+	TestDimensions
+	Type      MetricDataType
+	Timestamp uint64
+	Value     float64
+}
+
+// AssertTranslatorMap asserts that some OTLP data is mapped into some other Datadog data by a given translator.
+// OTLP data and Datadog data are stored in separate files passed as arguments.
+// The Datadog data base filename must start with the OTLP data base filename possibly followed by the translator options.
+//
+// To generate OTLP data to be used on this assert, use the pmetric.JSONMarshaler and json.Indent.
+// If the Datadog data does not match, a file ending in .actual will be generated containing the actual translator output.
+func AssertTranslatorMap(t *testing.T, translator *Translator, otlpfilename string, datadogfilename string) {
+
+	// Check that the filenames follow conventions.
+	prefix := strings.TrimSuffix(filepath.Base(otlpfilename), ".json")
+	if !strings.HasPrefix(filepath.Base(datadogfilename), prefix) {
+		t.Errorf("%q and %q do not follow prefix convention", otlpfilename, datadogfilename)
+		return
+	}
+
+	// Unmarshal OTLP data.
+	otlpbytes, err := os.ReadFile(otlpfilename)
+	require.NoError(t, err, "failed to read OTLP file %q", otlpfilename)
+
+	unmarshaler := &pmetric.JSONUnmarshaler{}
```

Comment:
Just my 2c but I think
```go
var unmarshaler pmetric.JSONUnmarshaler
```
Is nicer, cleaner and will work equally well. Same for all the other scenarios below.

Reply:
Done